### PR TITLE
Add password change to user controler

### PR DESCRIPTION
**Pull request by Unraveler**
Added change password method to User Controller
Added translation strings for error messages
Added PasswordChange to DTO

Please review @dnlkoch

### CODE REVIEW

File: shogun-lib/src/main/java/de/terrestris/shoguncore/controller/UserController.java [hunk 1]
```diff
@@ -109,4 +116,9 @@ public RedirectView confirmRegistration(@RequestParam("token") String token, Htt
         return redirectView;
     }
 
+    @PostMapping(value = "/password/change/{userId}")
+    public void changePassword(@PathVariable("userId") Long userId, @RequestBody PasswordChange passwordChangeBody) {
+            service.changeUserPassword(userId, passwordChangeBody);
```
Comment (dnlkoch) [inline shogun-lib/src/main/java/de/terrestris/shoguncore/controller/UserController.java:121]:
```suggestion
        service.changeUserPassword(userId, passwordChangeBody);
```

File: shogun-lib/src/main/java/de/terrestris/shoguncore/service/UserService.java [hunk 2]
```diff
@@ -150,6 +158,45 @@ private void enableUser(User user) {
...
+            throw new ResponseStatusException(
+                HttpStatus.BAD_REQUEST,
+                messageSource.getMessage(
+                    "password.change.USER_NOT_FOUND_ERROR",
+                    null,
+                    LocaleContextHolder.getLocale()
+                )
+            );
+        }
+    }
```
Comment (dnlkoch) [inline shogun-lib/src/main/java/de/terrestris/shoguncore/service/UserService.java:199]: The error handling (while throwing HTTP status exceptions) should take place in the controller.

**Review by dnlkoch — COMMENTED** [review]
Very useful addition, thanks @Unraveler!

**Comment by weskamm** [comment]
I dont see any checks that the user requesting the change is the same as the id that was given. Does this mean anyone can change the password of someone else? Or is this secured through repositorys `findById` method?

**Comment by Unraveler** [comment]
@weskamm , this was being handled in the service. Now it is being handled in the controller

**Comment by Unraveler** [comment]
@weskamm I did not correctly understood what you mean, Daniel clarified me. It should be ok now

File: shogun-lib/src/main/java/de/terrestris/shoguncore/service/UserService.java [hunk 3]
```diff
@@ -158,31 +158,22 @@ private void enableUser(User user) {
...
      * Changes user Password
      * @param user the user that is requesting a password change
-     * @param passwordChangeBody the request body, containing the old and new password
+     * @param passwordChange object containing the old and new password
      * @return
      */
-    public void changeUserPassword(User user, PasswordChange passwordChangeBody) {
-        String newPassword = passwordEncoder.encode(passwordChangeBody.getNewPassword());
-        user.setPassword(newPassword);
+    @PreAuthorize("hasRole('ROLE_ADMIN') or hasPermission(#user, 'UPDATE')")
```
Comment (weskamm) [inline shogun-lib/src/main/java/de/terrestris/shoguncore/service/UserService.java:164]: I like :-)

File: .github/workflows/v3.yml [hunk 4]
```diff
@@ -8,16 +8,16 @@ on:
...
     steps:
       - name: Checkout sources
         uses: actions/checkout@v2.0.0
 
       - name: Set up Java 11
-        uses: actions/setup-java@v1.3.0
+        uses: actions/setup-java@v1.4.3
         with:
-          java-version: 11.0.6
+          java-version: 11.0.8
```
Comment (dnlkoch) [inline .github/workflows/v3.yml:20]:
```suggestion
          java-version: 11.0.9
```

File: shogun-lib/src/main/java/de/terrestris/shoguncore/controller/UserController.java [hunk 5]
```diff
@@ -32,6 +39,15 @@
     @Autowired
     private Environment env;
 
+    @Autowired
+    private UserRepository userRepository;
```
Comment (dnlkoch) [inline shogun-lib/src/main/java/de/terrestris/shoguncore/controller/UserController.java:43]: Can be removed.

File: shogun-lib/src/main/java/de/terrestris/shoguncore/controller/UserController.java [hunk 6]
```diff
@@ -32,6 +39,15 @@
...
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
```
Comment (dnlkoch) [inline shogun-lib/src/main/java/de/terrestris/shoguncore/controller/UserController.java:46]: Can be removed as well.

**Review by dnlkoch — APPROVED** [review]

**Pull request merged**